student/tests.py

```diff
@@ -92,3 +92,12 @@ def test_if_user_is_a_student_returns_true_if_the_auth_user_is_associated_with_a
     example_user_not_student.save()
     assert Student.is_student(example_user_not_student.id) is False
     assert Student.is_student(example_student_a.user.id) is True
+
+
+@pytest.mark.django_db
+def test_get_user_as_student_object(example_student_a):
+    example_user_not_student = User.objects.create_user("notstudent", "password")
+    example_user_not_student.save()
+    with pytest.raises(Student.DoesNotExist):
+        Student.get_student(example_user_not_student.id)
```

Comment:
how come it's not exist it you created it on line 99?

Reply:
I created a user that is specifically **not** a student in line 99. The "Student.DoesNotExist" exception means that there is no **Student** with that user id in the database but it doesn't mean that it doesn't exist as a user.
Example_student_a is a student though because that's how he was created in line 27.

Reply:
okay but why are you creating it if you don't need it for the test (line 103) ?

Reply:
I am using it for a test. There are 2 tests in this function:
1. In line 103 I am testing the get_student() function with a user that **is** a Student instance.
2. In lines 101-102 I am testing the get_student() function with a user that is **not** a Student instance. In this case the function should throw a "Student.DoesNotExist" exception, I am testing that it indeed throws that exception. This is the syntax for testing exceptions according to what I read here https://www.kite.com/python/answers/how-to-assert-an-exception-is-raised-using-pytest-in-python
If for example in line 102 I write "Student.get_student(example_student_a)"  then the test will fail because there **is** such a student therefore an exception is **not** thrown.

Reply:
Should I just remove this test?

Reply:
okay I get it now, so this is kind of a negative testing which is essentially another test. i.e `test_get_student_raises_exception_for_non_student_user`

Reply:
OK I changed it into 2 seperate tests as you suggested. Although I must say that I feel like this PR is being dragged on way longer than it should be. I can understand if this was a conversation we have in one day but each day you give me a different comment and at this rate it will be very hard for us to be ready for the demo on Sunday.

Reply:
@eldaraharonson  Thank you for sharing this, sometimes it happens that people find out new things on the second review and so it gets more time for the pr to get merged. It's not intentional obviously but these things happen and it's good that you speak up when it happens.